pkg/oci/oci.go

```diff
@@ -126,12 +122,32 @@ func getTimeFromAnnotations(annotations map[string]string) string {
 
 // PullGadgetImage pulls the gadget image into the local oci store and returns its descriptor.
 func PullGadgetImage(ctx context.Context, image string, authOpts *AuthOptions) (*GadgetImageDesc, error) {
+	var desc *GadgetImageDesc
+	err := retry(func() error {
+		var err error
+		desc, err = pullGadgetImage(ctx, image, authOpts)
+		return err
+	})
+
+	return desc, err
+}
+
+func pullGadgetImage(ctx context.Context, image string, authOpts *AuthOptions) (*GadgetImageDesc, error) {
 	ociStore, err := getLocalOciStore()
 	if err != nil {
 		return nil, fmt.Errorf("getting oci store: %w", err)
 	}
 
-	return pullGadgetImageToStore(ctx, ociStore, image, authOpts)
+	desc, err := pullGadgetImageToStore(ctx, ociStore.Store, image, authOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := ociStore.SaveIndexWithLock(); err != nil {
```

Comment:
I don't see how this PR solves https://github.com/inspektor-gadget/inspektor-gadget/pull/3842#discussion_r1906998862

`getLocalOciStore` reads the oci store locally in a syncronous manner itself. But after the call another process can read the store too. If now both pull different images and save their indexes wouldn't we miss an image in the store index? (See linked comment)

Reply:
> If now both pull different images and save their indexes wouldn't we miss an image in the store index?

The `SaveIndexWithLock` function checks if the index was changed by another process before writing to it, if it was changed, it returns `ErrRetry` forcing the operation to be retried again.

Reply:
Somehow missed the DeepEqual +1